PrototypeDesignPattern/PrototypeDesign.java

```diff
@@ -0,0 +1,73 @@
+package PrototypeDesignPattern;
+
+public class PrototypeDesign {
+    public static void main(String[] args) {
+        GhostFreak g = new GhostFreak();
+        g.isAlien = true;
+        g.name = "G";
+        g.volume = 10.10;
+        GhostFreak g2 = (GhostFreak)g.clone();
+        System.out.println(g.isAlien + " " + g.name + " " + g.volume);
+        System.out.println(g2.isAlien + " " + g2.name + " " + g.volume);
+
+        Fourarms f = new Fourarms();
+        f.isAlien = true;
+        f.name = "F";
+        f.volume = 1.1;
+        Fourarms f2 =(Fourarms)f.clone();
+        System.out.println(f.isAlien + " " + f.name + " " + f.volume);
+        System.out.println(f2.isAlien + " " + f2.name + " " + f2.volume);
+
+    }
+}
+
+abstract class BenAlienForce {
+    boolean isAlien;
+    String name;
+
+    BenAlienForce() {
+    }
+
+    BenAlienForce(BenAlienForce b) {
+        this.isAlien = b.isAlien;
+        this.name = b.name;
+    }
+
+    public abstract BenAlienForce clone();
+}
+
+class GhostFreak extends BenAlienForce {
+    Double volume;
```

Comment:
[double vs Double](https://stackoverflow.com/questions/20437003/what-is-the-difference-between-double-and-double-in-java)

Do you really need Double?